core/core-impl/src/main/java/ai/timefold/solver/core/impl/solver/scope/SolverScope.java

```diff
@@ -224,6 +226,22 @@ public long getTimeMillisSpent() {
         return endingSystemTimeMillis - startingSystemTimeMillis;
     }
 
+    public ProblemStatistics getProblemStatistics() {
+        return problemStatistics;
+    }
+
+    public void setProblemStatistics(ProblemStatistics problemStatistics) {
+        this.problemStatistics = problemStatistics;
+    }
+
+    public ProblemStatistics calculateProblemStatistics(Solution_ problem) {
+        var solutionDescriptor = getSolutionDescriptor();
+        return new ProblemStatistics(solutionDescriptor.getGenuineEntityCount(problem),
+                solutionDescriptor.getGenuineVariableCount(problem),
+                solutionDescriptor.getMaximumValueRangeSize(problem),
+                solutionDescriptor.getProblemScale(problem));
+    }
```

Comment:
This might as well be on the `SolutionDescriptor` directly. It only uses stuff from there.